renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -525,6 +580,109 @@ struct VulkanPixelHistoryCallback : public VulkanDrawcallCallback
     return replacements;
   }
 
+  uint32_t GetEventFlags(const VulkanCreationInfo::Pipeline &p, const VulkanRenderState &pipestate)
+  {
+    uint32_t flags = 0;
+
+    // Culling
+    {
+      if(p.cullMode != VK_CULL_MODE_NONE)
+        flags |= TestEnabled_Culling;
+
+      if(p.cullMode == VK_CULL_MODE_FRONT_AND_BACK)
+        flags |= TestMustFail_Culling;
+    }
+
+    // Depth and Stencil tests.
+    {
+      if(p.depthBoundsEnable)
+        flags |= TestEnabled_DepthBounds;
+
+      if(p.depthTestEnable)
+      {
+        if(p.depthCompareOp != VK_COMPARE_OP_ALWAYS)
+          flags |= TestEnabled_DepthTesting;
+        if(p.depthCompareOp == VK_COMPARE_OP_NEVER)
+          flags |= TestMustFail_DepthTesting;
+      }
+
+      if(p.stencilTestEnable)
+      {
+        if(p.front.compareOp != VK_COMPARE_OP_ALWAYS || p.back.compareOp != VK_COMPARE_OP_ALWAYS)
+          flags |= TestEnabled_StencilTesting;
+
+        if(p.front.compareOp == VK_COMPARE_OP_NEVER && p.back.compareOp == VK_COMPARE_OP_NEVER)
+          flags |= TestMustFail_StencilTesting;
+        else if(p.front.compareOp == VK_COMPARE_OP_NEVER && p.cullMode == VK_CULL_MODE_BACK_BIT)
+          flags |= TestMustFail_StencilTesting;
+        else if(p.cullMode == VK_CULL_MODE_FRONT_BIT && p.back.compareOp == VK_COMPARE_OP_NEVER)
+          flags |= TestMustFail_StencilTesting;
+      }
+    }
+
+    // Scissor
+    {
+      bool inRegion = false;
+      bool inAllRegions = true;
+      // Do we even need to know viewerport here?
+      const VkRect2D *pScissors;
+      uint32_t scissorCount;
+      if(p.dynamicStates[VkDynamicViewport])
+      {
+        pScissors = pipestate.scissors.data();
+        scissorCount = (uint32_t)pipestate.scissors.size();
+      }
+      else
+      {
+        pScissors = p.scissors.data();
+        scissorCount = (uint32_t)p.scissors.size();
+      }
+      for(uint32_t i = 0; i < scissorCount; i++)
+      {
+        const VkOffset2D &offset = pScissors[i].offset;
+        const VkExtent2D &extent = pScissors[i].extent;
+        if((m_X >= (uint32_t)offset.x) && (m_Y >= (uint32_t)offset.y) &&
+           (m_X < (offset.x + extent.width)) && (m_Y < (offset.y + extent.height)))
+          inRegion = true;
+        else
+          inAllRegions = false;
+      }
+      if(!inRegion)
+        flags |= TestMustFail_Scissor;
+      if(inAllRegions)
+        flags |= TestMustPass_Scissor;
+    }
+
+    // Blending
+    {
+      if(m_pDriver->GetDeviceFeatures().independentBlend)
+      {
+        for(size_t i = 0; i < p.attachments.size(); i++)
+        {
+          if(p.attachments[i].blendEnable)
+          {
+            flags |= Blending_Enabled;
+            break;
+          }
+        }
+      }
+      else
+      {
+        // Might not have attachments if rasterization is disabled?
```

Comment:
I think this comment is accurate - you should add a check that `p.attachments` is non-empty.

Reply:
Added.